src/lua/api-gateway/validation/oauth2/oauthTokenValidator.lua

```diff
@@ -210,14 +210,17 @@ function _M:validateOAuthToken()
             self:setKeyInLocalCache(cacheLookupKey, cachedToken, local_expire_in, "cachedOauthTokens")
             self:setContextProperties(obj)
             return ngx.HTTP_OK
+        elseif (tokenValidity == 0) then
```

Comment:
This case should be treaded separately as rounding to seconds, will trigger a scenario where the token is still in the cache for 1 second, but the token validity is considered to be 0.